src/main/java/BankingSystem.java

```diff
@@ -0,0 +1,163 @@
+import card.Card;
+import card.CardGenerator;
+import dao.Account;
+import dao.AccountDao;
+import dao.Dao;
+
+import java.util.Scanner;
+
+public class BankingSystem {
+
+    private final Dao<Account> dao;
+    private final CardGenerator cardGenerator;
+    private final Scanner scanner;
+    private Account currentAccount;
+
+    public BankingSystem(String dbName) {
+        dao = new AccountDao(dbName);
+        cardGenerator = new CardGenerator();
+        scanner = new Scanner(System.in);
+    }
+
+    public void showMenu() {
+        while (true) {
+            welcomeMenu();
+
+            String input = scanner.next();
+
+            switch (input) {
+                case "0" -> System.exit(0);
+                case "1" -> registerAccount();
+                case "2" -> logIntoAccount();
+                default -> System.out.println("You've entered invalid menu item.\n");
+            }
+        }
+    }
+
+    private void welcomeMenu() {
+        System.out.println("""
+                1. Create an account
+                2. Log into account
+                0. Exit""");
+    }
+
+    private void registerAccount() {
+        Card card = cardGenerator.generateCredentials();
+        dao.save(new Account(card, 0));
+        System.out.printf("""
+                Your card has been created
+                Your card number:
+                %s
+                Your card PIN:
+                %s""", card.number(), card.pin());
+
+        System.out.println();
+        System.out.println();
+    }
+
+    private void logIntoAccount() {
+        System.out.println("\nEnter your card number:");
+        String inputtedCardNumber = scanner.next();
+        System.out.println("Enter your PIN:");
+        String inputtedCardPIN = scanner.next();
+        boolean isValid = accountValidation(inputtedCardNumber, inputtedCardPIN);
+
+        if (isValid) {
+            System.out.println("\nYou have successfully logged in!\n");
+            dao.get(inputtedCardNumber, inputtedCardPIN).ifPresent(account -> currentAccount = account);
+            accountMenu();
+        } else {
+            System.out.println("\nWrong card number or PIN!\n");
+        }
```

Comment:
Can you use another Optional method `ifPresentOrElse()`? it should match better here.